core/kernel/dt.c

```diff
@@ -553,23 +554,36 @@ void init_external_dt(unsigned long phys_dt)
 	if (!IS_ENABLED(CFG_EXTERNAL_DT))
 		return;
 
-	if (!phys_dt) {
+	if (IS_ENABLED(CFG_TRANSFER_LIST))
+		/* get DTB from the mapped TL */
+		fdt = transfer_list_entry_data
+			(transfer_list_find(dt->tl, TL_TAG_FDT));
```

Comment:
What is the motivation for this suggestion? And `transfer_list_find()` will return a `transfer_list_entry *` but not `transfer_list_header *`

Reply:
Rational regarding the braces is that the inline comment + the instruction make the block a multi line block so prefer have braces:
```c
	if (IS_ENABLED(CFG_TRANSFER_LIST)) {
		/* get DTB from the mapped TL */
		fdt = transfer_list_entry_data
			(transfer_list_find(dt->tl, TL_TAG_FDT));
	}
```

Rational for an extra local variable is to prevent the line above break,
and comply with 80char/line max. Or could break line after the function opening parenthesis:
```c
	if (IS_ENABLED(CFG_TRANSFER_LIST)) {
		/* get DTB from the mapped TL */
		fdt = transfer_list_entry_data(
			transfer_list_find(dt->tl, TL_TAG_FDT));
	}
```

Reply:
'make checkpatch' complains 'CHECK: Lines should not end with a '('' for your second suggestion.
I think it is fair enough to just remove the inline comment.